src/main/drivers/system_stm32f4xx.c

```diff
@@ -39,18 +39,39 @@ void systemReset(void)
     NVIC_SystemReset();
 }
 
+PERSISTENT uint32_t bootloaderRequest = 0;
+
 void systemResetToBootloader(void)
 {
     if (mpuResetFn) {
         mpuResetFn();
     }
 
-    *((uint32_t *)0x2001FFFC) = 0xDEADBEEF; // 128KB SRAM STM32F4XX
+    bootloaderRequest = 1;
 
     __disable_irq();
     NVIC_SystemReset();
 }
 
+#define SYSTEM_MEMORY __attribute__((section(".system_isr_vector"), aligned(4)))
+typedef void resetHandler_t(void);
+
+// System memory interrupt vector table
+static SYSTEM_MEMORY uint32_t           const volatile  systemStackEnd;
+static SYSTEM_MEMORY resetHandler_t *   const volatile  systemResetHandler;
+
+void checkForBootLoaderRequest(void)
+{
+    if (bootloaderRequest == 1) {
+        bootloaderRequest = 0;
+
+        __enable_irq();
```

Comment:
BTW: isn't IRQs disabled after reset?

Reply:
That's an interesting question, AN2606 says specifically that before jumping to Bootloader from user code one should disable IRQs, while our code does enable them for some reason.
I've tried disabling them explicitly and it lead to non-functional reboot to bootloader, not I had no stlink at hand to debug yet.

Reply:
Seems like `__enable_irq()` is not necessary there, as IRQs are indeed disabled at this stage (right after reboot). But why `disable_irq()` leads to failure? No idea without a debugger.